nemo_skills/code_execution/sandbox.py

```
@@ -157,40 +157,97 @@ async def execute_code(
-            # TODO: this history restoration is not ideal, we are returning output that is not the actual output of the new code
-            # and the execution of the concatenation may not be the same as the execution of the respective cells
-            # see details in this thread: https://github.com/NVIDIA/NeMo-Skills/pull/803#discussion_r2338240505
-            history = self.session_histories.get(session_id, [])
-            combined_code = "\n".join(history) + ("\n" if history else "") + generated_code
-            request = self._prepare_request(
-                combined_code, timeout, language, std_input, max_output_characters, traceback_verbosity
+            history = list(self.session_histories.get(session_id_str, []))
+            if request_session_id_str is not None:
+                try:
+                    await self.delete_session(request_session_id_str)
+                except Exception as exc:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)